arch/arm/src/nrf52/Kconfig

```diff
@@ -633,4 +633,16 @@ config NRF52_I2C_MASTER_DISABLE_NOSTART
 	bool "Disable the I2C Master NOSTART flag support"
 	default n
 
+config NRF52_I2C_MASTER_COPY_BUF_SIZE
+	int "Static buffer size for NOSTART flag support"
+	depends on !NRF52_I2C_MASTER_DISABLE_NOSTART
+	default 4
+	---help---
+		To combine two i2c messages that are part of a
+		single transaction (NO_STOP-NO_START) the nrf52
+		hardware requires these be joined into a single
+		transfer. This static buffer will be used if the
+		transaction will fit otherwise it will fall back
+		on malloc.
```

Comment:
Could you move part of this explanation to `NRF52_I2C_MASTER_DISABLE_NOSTART` (to explain why this option exists)?

Reply:
I added help text to that option as well including the risk.  I left this help text on this option so people can still have the context when configuring.